src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -33,86 +20,161 @@
+    public void setCoberturaChkBx(CoberturaChkBx coberturaChkBx) {
         this.coberturaChkBx = coberturaChkBx;
+        this.coberturaReportFilePath = this.coberturaChkBx.getCoberturaReportFilePath();
     }
-
+    
     @DataBoundSetter
-    public void setStmResultsChkBx(boolean stmResultsChkBx) {
+    public void setStmResultsChkBx(StmResultsChkBx stmResultsChkBx) {
         this.stmResultsChkBx = stmResultsChkBx;
+        this.stmResultsFilePath = this.stmResultsChkBx.getStmResultsFilePath();
     }
-
+    
     @DataBoundSetter
-    public void setModelCoverageChkBx(boolean modelCoverageChkBx) {
+    public void setModelCoverageChkBx(ModelCovChkBx modelCoverageChkBx) {
         this.modelCoverageChkBx = modelCoverageChkBx;
-    }
+        this.modelCoverageFilePath = this.modelCoverageChkBx.getModelCoverageFilePath();
+    }   
 
     @DataBoundSetter
-    public void setPdfReportChkBx(boolean pdfReportChkBx) {
+    public void setPdfReportChkBx(PdfChkBx pdfReportChkBx) {
         this.pdfReportChkBx = pdfReportChkBx;
+        this.pdfReportFilePath = this.pdfReportChkBx.getPdfReportFilePath();
     }
-
-    public boolean getTapChkBx() {
-        return tapChkBx;
+    
+    public String getTapReportFilePath() {
+        return this.tapReportFilePath;
+    }      
+    public boolean getIsPdfChecked() {
+        if(this.pdfReportChkBx != null) {
+            return true;
+        }
+        return false;
     }
-
-    public boolean getJunitChkBx() {
-        return junitChkBx;
+    public TapChkBx getTapChkBx() {
+        return this.tapChkBx;
     }
-
-    public boolean getCoberturaChkBx() {
-        return coberturaChkBx;
+    
+    public boolean getIsTapChecked() {
+        if(this.tapChkBx != null) {
+            return true;
+        }
+        return false;
     }
-
-    public boolean getStmResultsChkBx() {
-        return stmResultsChkBx;
+    
+    public JunitChkBx getJunitChkBx() {
+        return this.junitChkBx;
     }
-
-    public boolean getModelCoverageChkBx() {
-        return modelCoverageChkBx;
+    
+    public String getJunitReportFilePath() {
+        return this.junitReportFilePath;
     }
-
-    public boolean getPdfReportChkBx() {
-        return pdfReportChkBx;
+    
+    public boolean getIsJunitChecked() {
+        if(this.junitChkBx != null) {
+            return true;
+        }
+        return false;
     }
-
+    
+    public CoberturaChkBx getCoberturaChkBx() {
+        return this.coberturaChkBx;
+    }
+    
+    public String getCoberturaReportFilePath() {
+        return this.coberturaReportFilePath;
+    }
+    
+    public boolean getIsCoberturaChecked() {
+        if(this.coberturaChkBx != null) {
+            return true;
+        }
+        return false;
+    }
+      
+    public StmResultsChkBx getStmResultsChkBx() {
+        return this.stmResultsChkBx;
+    } 
+    
+    public String getStmResultsFilePath() {
+        return stmResultsFilePath;

```

Comment:
Just curious, why is there no "this" over here?

Reply:
it was missed :) thanks for pointing it out